components/script/stylesheet_loader.rs

```diff
@@ -213,6 +202,187 @@ impl FetchResponseListener for StylesheetContext {
         }
     }
 
+    fn parse_async(&mut self, status: Result<ResourceFetchTiming, NetworkError>) {
+        let thread_pool = match STYLE_THREAD_POOL.style_thread_pool.as_ref() {
+            Some(thread_pool) => thread_pool,
+            // If we don't have a thread pool, fall back to sync parsing.
+            None => return self.parse_sync(status),
+        };
+
+        let elem = self.elem.root();
+        let document = self.document.root();
+
+        if status.is_ok() {
+            let metadata = match self.metadata.take() {
+                Some(meta) => meta,
+                None => return,
+            };
+            let is_css = metadata.content_type.map_or(false, |ct| {
+                let mime: Mime = ct.into_inner().into();
+                mime.type_() == mime::TEXT && mime.subtype() == mime::CSS
+            });
+            let data = if is_css {
+                mem::replace(&mut self.data, vec![])
+            } else {
+                vec![]
+            };
+
+            // TODO: Get the actual value. http://dev.w3.org/csswg/css-syntax/#environment-encoding
+            let environment_encoding = UTF_8;
+            let final_url = metadata.final_url;
+            // FIXME: Revisit once consensus is reached at:
+            // https://github.com/whatwg/html/issues/1142
+            let successful = metadata.status.map_or(false, |(code, _)| code == 200);
```

Comment:
It used to be consistent, in that error event was fired in the `FinishAsyncStylesheetLoadTask` in all cases, but it seems (asynchronously) firing this event there, on `status.is_err()`, caused the timeouts. I haven't had the time to figure out exactly why.
I don't like the inconsistency either, but the difficulty I see is that in the sync version, this is determined *after* parsing and setting the sheet, so if we wanted consistency, we'd have to go on the parser thread only to send a task to script to fire the error event, which causes timeouts and what my latest commit fixes.
(I tried if only parsing the sheet if `status.is_ok() && metadata.status.map_or(false, |(code, _)| code == 200)` was true worked, because it would simplify the branching logic, but this also caused timeouts.)

Reply:
No worries, I was also afk for the holidays :) Happy new year btw! I haven't used it yet, but I'll try it sometime in the next few days.